smoderp2d/time_step.py

```diff
@@ -1,265 +1,354 @@
 # @package smoderp2d.time_step methods to perform
 #  time step, and to store intermediate variables
 
+from itertools import cycle
+from uu import Error
+from matplotlib.pylab import norm
 from smoderp2d.core.general import Globals, GridGlobals
 import smoderp2d.processes.rainfall as rain_f
-import smoderp2d.processes.infiltration as infilt
+import smoderp2d.processes.infiltration as infiltration
+import smoderp2d.processes.rill as rill
+from smoderp2d.core.surface import inflows_comp, surface_retention_impl
+from smoderp2d.core.surface import surface_retention_update
+from smoderp2d.core.surface import update_state1
+from smoderp2d.core.surface import update_state
+
+from smoderp2d.exceptions import NegativeWaterLevel
 
 import numpy as np
 import numpy.ma as ma
+import scipy as sp
+import scipy.optimize as spopt
 
 
-from smoderp2d.core.surface import runoff
-from smoderp2d.core.surface import surface_retention
+from smoderp2d.core.surface import sheet_runoff
+from smoderp2d.core.surface import rill_runoff
+from smoderp2d.core.surface import compute_h_hrill
+from smoderp2d.core.surface import compute_h_rill_pre
 
 
 # Class manages the one time step operation
-#
-#  the class also contains methods to store the important arrays to reload that
-#  if the time step is adjusted
-#
-class TimeStep:
-    """TODO."""
```

Comment:
Was it necessary to delete the TODO without replacing with a real docstring?

Reply:
It wasn't. Probably it was my superdestructive editing in the begging of the work on this project.

Reply:
Change took back by one of us some time ago (not sure whom, not sure which commit anymore). Thanks.